python/sglang/srt/distributed/device_communicators/shm_broadcast.py

```diff
@@ -225,8 +225,10 @@ def __init__(
             remote_subscribe_port = get_open_port()
             if is_valid_ipv6_address(connect_ip):
                 self.remote_socket.setsockopt(IPV6, 1)
-                connect_ip = f"[{connect_ip}]"
-            socket_addr = f"tcp://{connect_ip}:{remote_subscribe_port}"
```

Comment:
Note: This change fixes IPv6 support for the distributed setup, rather than for the P/D setup -- but it feels appropriate to include it here as well.